lib/faraday.rb

```diff
@@ -112,6 +112,13 @@ def respond_to_missing?(symbol, include_private = false)
     # Internal: Proxies method calls on the Faraday constant to
     # .default_connection.
     def method_missing(name, *args, &block)
+      # When running under debugger with a breakpoint set,
+      # self.to_str is called repeatedly during module load,
+      # including the time before the default_connection getter below is loaded
+      if name == :default_connection
+        return nil
+      end
```

Comment:
Style opinion: These guards would look easier to read as a "suffix if", like

```ruby
return nil if name == :default_connection
```

Reply:
Sure. I don't really have a preference here. I first tried `name == :default_connection && return nil` but that turned out to be invalid syntax :frowning_face: